src/_guides/language/concurrency/index.md

```diff
@@ -0,0 +1,454 @@
+the Dart VM's use of these primitives
+is an implementation detail that this page doesn't discuss.
+
+Many Dart apps use only one isolate (the _main isolate_),
+but you can create additional isolates,
+enabling parallel code execution on multiple processor cores.
+
+{{site.alert.info}}
+  **Platform note:**
+  All apps can use async-await, `Future`, and `Stream`.
+  Isolates are implemented only on the [Dart Native platform][];
+  Dart web apps can use [web workers][] for similar functionality.
+{{site.alert.end}}
+
+[Dart Native platform]: /overview#platform
+[web workers]: https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers
+
+
+
+## Asynchrony types and syntax
+
+If you’re already familiar with `Future`, `Stream`, and async-await,
+then you can skip ahead to the [isolates section][].
+
+[isolates section]: #how-isolates-work
+
+
+### Future and Stream types
+
+The Dart language and libraries use `Future` and `Stream` objects to
+represent values to be provided in the future.
+For example, a promise to eventually provide an `int` value
+is typed as `Future<int>`.
+A promise to provide a series of `int` values
+has the type `Stream<int>`.
+
+As another example, consider the dart:io methods for reading files.
+The synchronous `File` method [`readAsStringSync()`][]
+reads a file synchronously,
+blocking until the file is either fully read or an error occurs.
+The method then either returns an object of type `String`
+or throws an exception.
+The asynchronous equivalent, [`readAsString()`][],
+immediately returns an object of type `Future<String>`.
+At some point in the future,
+the `Future<String>` completes with either a string value or an error.
+
+[`readAsStringSync()`]: {{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-io/File/readAsStringSync.html
+[`readAsString()`]: {{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-io/File/readAsString.html
+
+Why does it matter whether a method is synchronous or asynchronous?
+It matters because most apps need to do more than one thing at a time.
+For example, an app might start an HTTP request,
+but need to update its display or respond to user input
+before the HTTP request completes.
+Asynchronous code helps apps stay responsive.
+
+### The async-await syntax
+
+The `async` and `await` keywords provide
+a declarative way to define asynchronous functions
+and use their results.
+
+Here’s an example of some synchronous code that blocks while waiting for file I/O:
+
+```dart
+void main() {
+  // Read some data.
+  final fileData = _readFileSync();
+  final jsonData = jsonDecode(fileData);
+
+  // Use that data.
+  print('Number of JSON keys: ${jsonData.length}');
+}
+
+String _readFileSync() {
+  final file = File(filename);
+  final contents = file.readAsStringSync();
+  return contents.trim();
+}
+```
+
+Here’s similar code, but with changes (highlighted) to make it asynchronous:
+
+{% prettify dart tag=pre+code %}
+[!Future<void>!] main() [!async!] {
```

Comment:
This can stay `void`, since no one is going to call `main`: https://github.com/dart-lang/samples/pull/128#discussion_r764096958

```suggestion
void main() [!async!] {
```

Reply:
Ported to https://github.com/dart-lang/site-www/pull/3711